[ENH] Refactor count/countna to use FExpr (#3440)
h2oai/datatable: docs/api/dt/count.rst

.. xfunction:: datatable.count
    :src: src/core/expr/fexpr_count_countna.cc pyfn_count
    :cvar: doc_dt_count
    :tests: tests/test-reduce.py
    :signature: count(cols=None)

>     Calculate the number of non-missing values for each column from `cols`, if `cols` is provided, 
>     or the total number of rows if `cols` is not provided.

Reviewer: ```suggestion
    Calculate the number of non-missing values for each column from `cols`. When `cols` is not provided, 
    calculate the total number of rows.
```

    Parameters
    ----------
    cols: FExpr

>         Input columns. If no `cols` is passed, then the count of all rows is returned.

Reviewer: ```suggestion
        Input columns if any.
```

Reviewer: We don't need to mention here what is returned, as this is not a `return:` section.

    return: Expr
        f-expression having one row, and the same names and number of columns
        as in `cols`. All the returned column stypes are `int64`.
        If `cols` is not provided, the total number of rows 

>         (a combination of the count of missing and non-missing values) is returned.

Reviewer: Oh, by the way, could you please update the "See Also" sections for `count()` and `countna()` to point to each other? Should be pretty useful.
